model/CSVDialog.cc

```diff
@@ -149,14 +148,13 @@ std::string CSVDialog::loadWebFile(const std::string& url)
   if (found!=std::string::npos) {
       vector<unsigned char> zbuf(res.get().body().size());
       DeflateZStream zs(res.get().body(), zbuf);
-      zs.deflate();  
+      zs.deflate();
       
-      vector<char> cbuf(a85::size_for_a85(zs.total_out,false));
-      a85::to_a85(&zbuf[0],zs.total_out, &cbuf[0], false);
-      // this ensures that the escape sequence ']]>' never appears in the data
-      replace(cbuf.begin(),cbuf.end(),']','~');      
+      //vector<char> cbuf(zs.total_out);
+      //memcpy(&cbuf,&zbuf,zs.total_out);   
+      //memcpy(&cbuf[0],&zbuf[0],zbuf.size()*sizeof(zbuf[0]));      
     
-      outFile << cbuf.data();
+      outFile << zbuf.data();
```

Comment:
This will not work. Passing a char* to an output stream assumes that nul ('\0') terminates the data.
You will need to use the binary API of std::ostream, which I can't tell you off hand, you'll need to look it up.

Reply:
Oops, that was an obvious mistake, sorry. I first tried a memcpy(), but of course a container of size z.total_out will not be compatible with one of res.get().body(), so I got memory leaks. I thought I may have done something silly which could be fixed in a single memcpy() statement, say. But I will work through https://www.zlib.net/zlib_how.html and http://www.cplusplus.com/reference/ostream/ostream/ carefully. Thank you for the hint.